src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+        let address = unsafe { InetAddr::from_libc_sockaddr(info.ifa_addr) };
+        let netmask = unsafe { InetAddr::from_libc_sockaddr(info.ifa_netmask) };
+        let mut addr = InterfaceAddress {
+            interface_name: ifname.to_string_lossy().to_string(),
+            flags: InterfaceFlags::from_bits_truncate(info.ifa_flags as i32),
+            address: address,
+            netmask: netmask,
+            broadcast: None,
+            destination: None,
+        };
+
+        #[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
+        {
+            let ifo = unsafe { InetAddr::from_libc_sockaddr(info.ifa_ifu) };
+            if addr.flags.contains(IFF_POINTOPOINT) {
+                addr.destination = ifo;
+            } else if addr.flags.contains(IFF_BROADCAST) {
+                addr.broadcast = ifo;
+            }
+        }
+
+        #[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd",
+                    target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd",
+                    target_os = "bitrig", target_os = "solaris"))]
+        {
+            addr.destination = unsafe { InetAddr::from_libc_sockaddr(info.ifa_dstaddr) };
+        }
+
+        addr
+    }
+}
+
+/// Holds the results of `getifaddrs`.
+///
+/// Use the function `getifaddrs` to create this Iterator. Note that the
+/// actual list of interfaces can be iterated once and will be freed as
+/// soon as the Iterator goes out of scope.
+pub struct InterfaceAddressIterator {
+    base: *mut libc::ifaddrs,
+    next: *mut libc::ifaddrs,
+}
+
+impl Drop for InterfaceAddressIterator {
+    fn drop(&mut self) {
+        unsafe { libc::freeifaddrs(self.base) };
+    }
+}
+
+impl Iterator for InterfaceAddressIterator {
+    type Item = InterfaceAddress;
+    fn next(&mut self) -> Option<<Self as Iterator>::Item> {
+        match unsafe { self.next.as_ref() } {
+            Some(ifaddr) => {
+                self.next = ifaddr.ifa_next;
+                Some(InterfaceAddress::from_libc_ifaddrs(ifaddr))
+            }
+            None => None,
+        }
+    }
+}
+
+/// Get interface addresses using libc's `getifaddrs`
+///
+/// Note that the underlying implementation differs between implementations and
+/// the nix crate doesn't support `AF_PACKET` entries. Therefore, the returned
+/// list often contains empty entries.
+///
+/// # Example
+/// ```
+/// match nix::ifaddrs::getifaddrs() {
+///     Ok(ifaddrs) => for ifaddr in ifaddrs {
+///         match ifaddr.address {
+///             Some(address) => {
+///                 println!("interface {} address {}",
+///                     ifaddr.interface_name, address.to_std());
+///             },
+///             None => {}
+///         }
+///     },
+///     Err(err) => {
+///         println!("error getting interface addresses: {}", err)
+///     }
+/// }
+/// ```
+#[cfg(not(any(target_os = "android", target_os = "haiku")))]
```

Comment:
Can you check that this isn't on Android? According to [this](https://android.googlesource.com/platform/prebuilts/ndk/+/dev/platform/sysroot/usr/include/ifaddrs.h) it should.